packages/next-swc/crates/napi/src/app_structure.rs

```diff
@@ -106,21 +106,36 @@ async fn prepare_components_for_js(
     add(&mut map, project_path, "default", default).await?;
     add(&mut map, project_path, "route", route).await?;
     let mut meta = serde_json::value::Map::new();
```

Comment:
This should be a `#[derive(Serialize)]` data structure. This would make the code below more readable, type safe, and `add_meta` could be a method of that data structure.